torch/jit/_script.py

```diff
@@ -341,9 +341,14 @@ def unpackage_script_module(importer: PackageImporter, script_module_id: str) ->
     Called by ``torch.package.PackageImporter``'s Pickler's ``persistent_load`` function.
     Performs work of loading and returning a ScriptModule from a ``torch.package`` archive.
     """
+    assert(isinstance(importer.zip_reader, torch._C.PyTorchFileReader))
```

Comment:
TODO: I'm not sure how to handle the issue where TS objects can only be loaded when the `zip_reader` for a `PackageImporter` is a archive instead of a plain directory. This issue arrises because the TS deserialization code expects a `PyTorchFileReader`

Reply:
what does this mean? I thought `zip_reader` is always a PyTorchFileReader or a MockZipReader.

Reply:
Yeah it's confusing. The MockZipReader isn't a PyTorchFileReader reader and The TorchScript deserialization code expects an actual PyTorchFileReader

Reply:
I added a better error message. I think we can fix this in a follow up PR